### ISSUE TITLE
Hoth became ‘drain’ status because of Kill task failed 

### ISSUE THREAD

**Issue by luoyuqi-lab** [gqqnbig/shine-cluster#166]
When I did regularly check, I found hoth became to drain and could not be used.

Use `sinfo -R` to check, got 'Kill task failed'
![hoth drail 2022 11 7](https://user-images.githubusercontent.com/57832734/200304519-cd006f20-22c9-4845-af1c-5cddb904e401.png)

I check the log, in `/var/log/slurmctld.log`
![hoth kill job](https://user-images.githubusercontent.com/57832734/200304930-1359a295-d057-4dcc-b52e-1d6a0e6e10c0.png)

I‘m suspicious of hoth, because a few days ago, suddenly shared path mount failed also occur in hoth. Maybe there are some flaws when we rebuilt it a few months ago.

**Comment by luoyuqi-lab** [comment]
Now hoth is normal, by `sudo scontrol update nodename=hoth state=resume`.

Need more check for hoth.

[Reference](https://bugs.schedmd.com/show_bug.cgi?id=3941)
[slurmctld](https://slurm.schedmd.com/slurmctld.html)

**Comment by gqqnbig** [comment]
OK, just make sure the mounting will not fail again

**Comment by gqqnbig** [comment]
Well, so has the suspicious task been killed or not? You don't wanna leave
trash processes in the system.

Can #107 solve this problem? I guess even if you don't check servers, after
a number of days, the server reboots, and the problem should get away.

Or do you like to make a script that whenever a server goes drained, the
script runs your scontrol command?

In that case, the drained state seems useless to us. Can slurm completely
ignore the nuisance drain state?

**Comment by gqqnbig** [comment]
Before digging into this, remember to wrap up [high priority ones](https://github.com/gqqnbig/shine-cluster/labels/priority-high).

**Comment by luoyuqi-lab** [comment]
### Found the reason why compute nodes always drain:
1. Check logs in log in node aha (`/var/log/slurmctld.log`) and compute node dagobah (`/var/log/slurmd.log`). As mentioned before, in compute node, the kill task fails to cause it to become to `drain`. Task stuck by oom-kill.
![image](https://user-images.githubusercontent.com/57832734/235865937-e4f020bc-4aa4-493b-8a8b-bf265970a211.png)
![image](https://user-images.githubusercontent.com/57832734/235866268-3775f2a0-86c9-4947-b6d3-74b4e23d6760.png)

2. We found in new resumed dagobah, now no tasks, but the Swp is full, and so many things with Status ['D'](https://manpages.ubuntu.com/manpages/focal/en/man1/htop.1.html) (Uninterruptible sleep) occupied the VIRT. We even can not kill them manually.
![image](https://user-images.githubusercontent.com/57832734/235867183-942457a0-e339-453d-a30b-4267ddce890c.png)
![image](https://user-images.githubusercontent.com/57832734/235858568-5c798f44-1436-4e36-ab00-0fe4577ca174.png)
![image](https://user-images.githubusercontent.com/57832734/235858733-06493791-b0d9-4921-ace4-395c3c790f26.png)

3. We speculate wrong/improper codes from users cause some tasks [cannot be killed](https://bugs.schedmd.com/show_bug.cgi?id=3941), then the slurm stuck by `drain` status.

4. Solution: (1)Manually resume a drain node. (2)Write a script, auto-resume a drain node. (3) Regular reboot nodes to kill some uninterruptible sleep threads. (4)Stop give codes with bugs/flaws to the cluster.

Reference: [1](https://serverfault.com/questions/499301/what-is-the-danger-in-having-a-fully-used-swap-in-an-ubuntu-server) [2](https://serverfault.com/questions/132084/system-swapping-how-do-i-figure-out-what-the-problem-is?rq=1) [3](https://stackoverflow.com/questions/52421068/error-in-slurm-cluster-detected-1-oom-kill-events-how-to-improve-running-jo) [4](https://bugs.schedmd.com/show_bug.cgi?id=3941)

BTW, why `conda list --json` makes the node got stuck...... :cold_sweat:

**Comment by gqqnbig** [comment]
Well, interesting findings. 

Perhaps send out an email telling users not to run the conda command which has been run by a quite number of users. I don't even know how they found this peculiar command.

Also, ask them to check bugs in their python code. 

Do it right now, so Dagobah doesn't go drain again. What do you think?

**Comment by luoyuqi-lab** [comment]
> Well, interesting findings. 
> 
> Perhaps send out an email telling users not to run the conda command which has been run by a quite number of users. I don't even know how they found this peculiar command.
> 
> Also, ask them to check bugs in their python code. 
> 
> Do it right now, so Dagobah doesn't go drain again. What do you think?

I agree, but first I will find other commands that will stuck in the memory, I think not only `conda list --json`.
